modules/iam-github-oidc-role/main.tf

```
@@ -44,7 +44,8 @@ data "aws_iam_policy_document" "this" {
     condition {
       test     = "StringLike"
       variable = "${local.provider_url}:sub"
-      values   = [for subject in var.subjects : "repo:${subject}"]
+      # Strip `repo:` to normalize for cases where users may prepend it
+      values = [for subject in var.subjects : "repo:${replace(subject, "/^repo:/", "")}"]
```

```suggestion
      values = [for subject in var.subjects : "repo:${trimprefix(subject, "repo:")}"]
```

A bit shorter and nicer with native functions - https://developer.hashicorp.com/terraform/language/functions/trimprefix